src/main/java/io/fabric8/maven/docker/util/aws/AwsSdkAuthConfigFactory.java

```diff
@@ -0,0 +1,40 @@
+package io.fabric8.maven.docker.util.aws;
+
+import io.fabric8.maven.docker.access.AuthConfig;
+import io.fabric8.maven.docker.util.Logger;
+
+public class AwsSdkAuthConfigFactory {
+
+    private final Logger log;
+
+    public AwsSdkAuthConfigFactory(Logger log) {
+        this.log = log;
+    }
+
+    public AuthConfig createAuthConfig() {
+        try {
+            Class<?> credentialsProviderChainClass = Class.forName("com.amazonaws.auth.DefaultAWSCredentialsProviderChain");
+            Object credentialsProviderChain = credentialsProviderChainClass.getDeclaredConstructor().newInstance();
+            Object credentials = credentialsProviderChainClass.getMethod("getCredentials").invoke(credentialsProviderChain);
+            if (credentials == null) {
+                return null;
+            }
+
+            Class<?> sessionCredentialsClass = Class.forName("com.amazonaws.auth.AWSSessionCredentials");
+            String sessionToken = sessionCredentialsClass.isInstance(credentials)
+                    ? (String) sessionCredentialsClass.getMethod("getSessionToken").invoke(credentials) : null;
+
+            Class<?> credentialsClass = Class.forName("com.amazonaws.auth.AWSCredentials");
+            return new AuthConfig(
+                    (String) credentialsClass.getMethod("getAWSAccessKeyId").invoke(credentials),
+                    (String) credentialsClass.getMethod("getAWSSecretKey").invoke(credentials),
+                    "none",
+                    sessionToken
+            );
+        } catch (Throwable t) {
+            log.debug("Failed to fetch AWS credentials: %s", t);
```

Comment:
I would switch to a `warn` here as it already a Amazon SDK has been detected (so the intention is clear), but this failed. We can still move on with more authentication but I would expose this more prominently so that we eventually could fix that. E.g. this could happen when the SDK evolves ...